### PR TITLE
Tag beneficiaries contacts

### PR DESCRIPTION
**Pull request by IsaackMwenda**
This adds a boolean beneficiary column in analysis files to aid in quant and engagement analysis.

### CODE REVIEW

File: configuration/pipeline_config.json [hunk 1]
```diff
@@ -24,6 +24,11 @@
     "FirebaseCredentialsFileURL": "gs://avf-credentials/avf-id-infrastructure-firebase-adminsdk-6xps8-b9173f2bfd.json",
     "TableName": "OXFAM_KENYA_phone_number_avf_phone_id"
   },
+  "BeneficiaryFileURLs":[
+    "gs://avf-project-datasets/2020/OXFAM-WASH/oxfam_wash_kisumu_beneficiaries_deindentified.csv",
+    "gs://avf-project-datasets/2020/OXFAM-WASH/oxfam_wash_mombasa_beneficiaries_deindentified.csv",
+    "gs://avf-project-datasets/2020/OXFAM-WASH/oxfam_wash_nairobi_beneficiaries_deidentified.csv"
+    ],
```
Comment (as2388) [inline configuration/pipeline_config.json]: (nit: indentation)

File: fetch_raw_data.py [hunk 2]
```diff
@@ -156,6 +156,15 @@ def fetch_from_recovery_csv(user, google_cloud_credentials_file_path, raw_data_d
             TracedDataJsonIO.export_traced_data_iterable_to_jsonl(traced_runs, f)
         log.info(f"Exported TracedData")
 
+def fetch_beneficiary_file_csvs(google_cloud_credentials_file_path, pipeline_configuration, raw_data_dir):
+    for beneficiary_file_url in pipeline_configuration.beneficiary_file_urls:
+        beneficiary_file = beneficiary_file_url.split("/")[-1]
+
+        log.info(f"Saving '{beneficiary_file}' to file '{raw_data_dir}'...")
```
Comment (as2388) [inline fetch_raw_data.py]:
```suggestion
        log.info(f"Saving '{beneficiary_file}' to directory '{raw_data_dir}'...")
```
or even better:
```suggestion
        log.info(f"Saving '{beneficiary_file}' to file '{raw_data_dir}/{beneficiary_file}'...")
```

File: fetch_raw_data.py [hunk 3]
```diff
@@ -193,6 +202,10 @@ def main(user, google_cloud_credentials_file_path, pipeline_configuration_file_p
         else:
             assert False, f"Unknown raw_data_source type {type(raw_data_source)}"
 
+    # Fetch de-identified listening group CSVs
+    log.info(f"Fetching beneficiary files")
```
Comment (as2388) [inline fetch_raw_data.py]:
```suggestion
    log.info(f"Fetching beneficiary files...")
```

File: fetch_raw_data.py [hunk 4]
```diff
@@ -156,6 +156,15 @@ def fetch_from_recovery_csv(user, google_cloud_credentials_file_path, raw_data_d
             TracedDataJsonIO.export_traced_data_iterable_to_jsonl(traced_runs, f)
         log.info(f"Exported TracedData")
 
+def fetch_beneficiary_file_csvs(google_cloud_credentials_file_path, pipeline_configuration, raw_data_dir):
+    for beneficiary_file_url in pipeline_configuration.beneficiary_file_urls:
+        beneficiary_file = beneficiary_file_url.split("/")[-1]
+
+        log.info(f"Saving '{beneficiary_file}' to file '{raw_data_dir}'...")
+        with open(f'{raw_data_dir}/{beneficiary_file}', "wb") as beneficiary_output_file:
```
Comment (as2388) [inline fetch_raw_data.py:168]: If these files don't change, consider doing this only if the file doesn't exist to improve performance and reduce costs.

File: src/analysis_file.py [hunk 5]
```diff
@@ -1,19 +1,48 @@
...
 
 from src.lib import PipelineConfiguration, ConsentUtils
 from src.lib.configuration_objects import CodingModes
 
-
 class AnalysisFile(object):
+    @staticmethod
+    def tag_beneficiary_participants(user, data, pipeline_configuration, raw_data_dir):
+        """
+        This tags uids who are our partners beneficiaries.
```
Comment (as2388) [inline src/analysis_file.py:18]:
```suggestion
        This tags uids who are our partner's beneficiaries.
        
```

**Review by as2388 — APPROVED** [review]

File: configuration/pipeline_config.json (hunk 1 above)
Comment (IsaackMwenda) [reply configuration/pipeline_config.json]: updated thanks.

File: fetch_raw_data.py (hunk 2 above)
Comment (IsaackMwenda) [reply fetch_raw_data.py]: updated thanks.

File: fetch_raw_data.py (hunk 3 above)
Comment (IsaackMwenda) [reply fetch_raw_data.py]: updated thanks.

File: fetch_raw_data.py (hunk 4 above)
Comment (IsaackMwenda) [reply fetch_raw_data.py:168]: updated to download only if the file does not exist.

File: src/analysis_file.py (hunk 5 above)
Comment (IsaackMwenda) [reply src/analysis_file.py:18]: updated thanks.

**Pull request merged**